components/forms/EmailForm.tsx

```diff
@@ -1,182 +1,195 @@
-import React, { useState } from 'react'
-import styled, { css } from 'styled-components'
-import { addToMailchimp } from '../../utils'
-import { Input, Button } from '../ui'
-import ModalConfirmation from 'components/ui/ModalConfirmation'
-import useToggle from '../../utils/useToggle';
-import { useRouter } from 'next/router'
-
+import React, { useState } from 'react';
+import styled, { css } from 'styled-components';
+import { addToMailchimp } from '../../utils';
+import { Input, Button } from '../ui';
+import { useRouter } from 'next/router';
+import Image from 'next/image';
+import BettyWithLlama from '../../public/img/BettyWithLlama.png';
+import { ImCross } from "react-icons/im";
+import { IoIosWarning } from "react-icons/io";
+import { TiTick } from "react-icons/ti";
 
 interface EmailFormProps {
-  isFooter: boolean
+  isFooter: boolean;
 }
 
 export const EmailForm = (props: EmailFormProps) => {
-  const [email, setEmail] = useState('')
-  const [isEntering, setIsEntering] = useState(false)
+  const [email, setEmail] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [isEntering, setIsEntering] = useState(false);
   const [isProcessing, setIsProcessing] = useState(false);
 
-  const [isSuccessOpen, toggleSuccess] = useToggle(false);
-  const [isErrorOpen, toggleError] = useToggle(false);
-  const [isDuplicateOpen, toggleDuplicate] = useToggle(false);
-
+  const [successMessage, setSuccessMessage] = useState('');
+  const [warningMessage, setWarningMessage] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
```

Comment:
Not a fan of having 3 different fields.

Could you do something like this instead?

```tsx
  const [message, setMessage] = useState({ text: '', type: '' });

  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    setIsProcessing(true);
    setMessage({ text: '', type: '' });
    try {
      const result = await addToMailchimp(email, firstName, lastName);
      if (result.result === 'success') {
        setMessage({ text: "You've been added to the llama list.", type: 'success' });
      } else if (result.message === 'Bad Request') {
        setMessage({ text: "You're already in our herd!", type: 'warning' });
      } else {
        setMessage({ text: 'There was an error. Please try again.', type: 'error' });
      }
    } catch (error) {
      console.error('Error submitting email:', error);
      setMessage({ text: 'There was an error. Please try again.', type: 'error' });
    }
    setIsProcessing(false);
  };
```

Reply:
✅ done